techniques/applications/repoGpgKeyManagement/1.0/repoGpgKeyManagement.st

```diff
@@ -0,0 +1,142 @@
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+bundle agent check_repo_gpg_key
+{
+
+  vars:
+
+    &GPG_KEY_ACTION:{action |"repo_gpg_action[&i&]" string => "&action&";
+}&
+
+    &GPG_KEY_HASH,GPG_KEY_NAME:{hash,name |"repo_gpg_hash[&i&]" string => "&hash&", comment => "&name&";
+}&
+
+    &TRACKINGKEY:{uuid |"repo_gpg_uuid[&i&]" string => "&uuid&";
+}&
+
+    "repo_gpg_index" slist => getindices("repo_gpg_hash");
+
+&GPG_KEY_CONTENT:{content |"repo_gpg_content[&i&]" string => "&content&";
+}&
+
+    "repo_gpg_hash_lc[${repo_gpg_index}]" string => string_downcase("${repo_gpg_hash[${repo_gpg_index}]}"),
+      comment => "We need lower case to query RPM and it's also used in the apt-based module for class definition";
+
+   _stdlib_path_exists_apt_key::
+    "repo_gpg_hash_uc[${repo_gpg_index}]" string => string_upcase("${repo_gpg_hash[${repo_gpg_index}]}"),
+      comment => "To remove a key from APT we need the hash in explicit capital letters, lower/mixed case just does not work, even if it reports OK.";
+
+  classes:
+
+    "repo_gpg_key_${repo_gpg_index}_action_remove"
+      expression => strcmp("${repo_gpg_action[${repo_gpg_index}]}","del");
+
+    "repo_gpg_key_${repo_gpg_index}_action_add"
+      expression => strcmp("${repo_gpg_action[${repo_gpg_index}]}","add");
+
+  files:
+
+    "${g.rudder_var_tmp}/repo_gpg_key_${repo_gpg_index}_content"
+      comment       => "Create a temporary file for repo to import the GPG key. Unfortunately rpm does not accept /dev/stdin or - for importing, so we need this temporary.",
+      create        => "true",
+      edit_line     => insert_lines( "${repo_gpg_content[${repo_gpg_index}]}" ),
+      edit_defaults => ncf_empty_select("true"),
+      ifvarclass    => "repo_gpg_hashes_read_ok.!repo_gpg_key_${repo_gpg_hash_lc[${repo_gpg_index}]}_present.repo_gpg_key_${repo_gpg_index}_action_add",
+      classes       => classes_generic("repo_gpg_key_${repo_gpg_index}_file_created");
+
+    "${g.rudder_var_tmp}/repo_gpg_key_${repo_gpg_index}_content"
+      comment    => "Remove temporary file if the GPG key import has been tried, regardless of the outcome",
+      delete     => tidy,
+      ifvarclass => "repo_gpg_key_${repo_gpg_index}_imported_reached";
+
+  commands:
+
+   _stdlib_path_exists_apt_key::
+
+    "$(paths.apt_key) fingerprint | $(paths.sed) -rn 's/^pub\s+.{5}\/([A-F0-9]{8}) .+/\+repo_gpg_key_\1_present/gp;' | $(paths.tr) '[:upper:]' '[:lower:]'"
```

Comment:
Sorry to be nitpicky, but we have the habit of using ${ } rather than $( )
could you use ${paths.apt_key}